Modules/Scripted/SegmentStatistics/SegmentStatisticsCalculators/LabelmapSegmentStatisticsCalculator.py

```diff
@@ -0,0 +1,94 @@
+    self.defaultKeys = self.keys # calculate all measurements by default
+    #... developer may add extra options to configure other parameters
+
+  def computeStatistics(self, segmentID):
+    import vtkSegmentationCorePython as vtkSegmentationCore
+    requestedKeys = self.getRequestedKeys()
+
+    segmentationNode = slicer.mrmlScene.GetNodeByID(self.getParameterNode().GetParameter("Segmentation"))
+
+    if len(requestedKeys)==0:
+      return {}
+
+    containsLabelmapRepresentation = segmentationNode.GetSegmentation().ContainsRepresentation(
+      vtkSegmentationCore.vtkSegmentationConverter.GetSegmentationBinaryLabelmapRepresentationName())
+    if not containsLabelmapRepresentation:
+      return {}
+
+    segment = segmentationNode.GetSegmentation().GetSegment(segmentID)
+    segBinaryLabelName = vtkSegmentationCore.vtkSegmentationConverter.GetSegmentationBinaryLabelmapRepresentationName()
+    segmentLabelmap = segment.GetRepresentation(segBinaryLabelName)
+
+    # We need to know exactly the value of the segment voxels, apply threshold to make force the selected label value
+    labelValue = 1
+    backgroundValue = 0
+    thresh = vtk.vtkImageThreshold()
+    thresh.SetInputData(segmentLabelmap)
+    thresh.ThresholdByLower(0)
+    thresh.SetInValue(backgroundValue)
+    thresh.SetOutValue(labelValue)
+    thresh.SetOutputScalarType(vtk.VTK_UNSIGNED_CHAR)
+    thresh.Update()
+
+    #  Use binary labelmap as a stencil
+    stencil = vtk.vtkImageToImageStencil()
+    stencil.SetInputData(thresh.GetOutput())
+    stencil.ThresholdByUpper(labelValue)
+    stencil.Update()
+
+    stat = vtk.vtkImageAccumulate()
+    stat.SetInputData(thresh.GetOutput())
+    stat.SetStencilData(stencil.GetOutput())
+    stat.Update()
+
+    # Add data to statistics list
+    cubicMMPerVoxel = reduce(lambda x,y: x*y, segmentLabelmap.GetSpacing())
+    ccPerCubicMM = 0.001
+    stats = {}
+    if "Labelmap.voxel_count" in requestedKeys:
+      stats["Labelmap.voxel_count"] = stat.GetVoxelCount()
+    if "Labelmap.volume_mm3" in requestedKeys:
+      stats["Labelmap.volume_mm3"] = stat.GetVoxelCount() * cubicMMPerVoxel
+    if "Labelmap.volume_cc" in requestedKeys:
+      stats["Labelmap.volume_cc"] = stat.GetVoxelCount() * cubicMMPerVoxel * ccPerCubicMM
+    return stats
+
+  def getMeasurementInfo(self, key):
+    """Get information (name, description, units, ...) about the measurement for the given key"""
+    info = {}
+
+    # @fedorov could not find any suitable code.
+    # DCM has "Number of needles" etc., so probably "Number of voxels"
+    # should be added too. Need to discuss with @dclunie. For now, a
+    # QIICR private scheme placeholder.
+
+    info["Labelmap.voxel_count"] = \
+      self.generateMeasurementInfo(name="Voxel count", description="Number of voxels", units="voxels",
+                                   quantityCode=self.initCodedEntry("nvoxels", "99QIICR", "Number of voxels", True),
```

Comment:
OK, 111439 does not look like a convention that it would make sense to follow. "nvoxels" is not bad, as it is not displayed to users and it's very short, but it may be easier to maintain the coding scheme if code values are a bit more explicit, such as "NumberOfVoxels", "number_of_voxels", "voxel_count", ...

I just spend time with discussing this minor detail, as these codes may be very difficult to change in the future, so it is important to choose conventions that we can consistently adhere to. If we choose a very simplistic naming scheme (just lowercase letters, no separators) then it may be more difficult to find consistent and expressive code values for longer, more complex items.